### ISSUE TITLE
Add `plugin-config` package

### ISSUE THREAD

**Issue by chrispcampbell** [climateinteractive/SDEverywhere#229]
The earlier monolithic `sdeverywhere` package had support for reading csv config files and generating a simple web app that runs the generated model.  When we added the new build and plugin packages, I removed that config file / web app generation code from the core cli/compiler packages because it's the kind of thing that works better as an optional plugin (which also helps avoid having too many non-optional dependencies in the core packages).

Now that the build and plugin packages have been added, this would be a good time to add back some of that config file support as part of an optional `plugin-config` package.  The user would no longer need to manually implement the `modelSpec` function in the `sde.config.js` file, and could instead use this new package to read the csv files from a `config` directory.  This would supply the necessary info to allow the rest of the build process to proceed, and would also generate some JS/JSON config files that could be used in the "core" and "app" packages that will be created by the new feature in #228.

**Comment by chrispcampbell** [comment]
As part of implementing this issue, I added an `examples/sir` project that uses the new plugin-config package.  For now, it includes boilerplate "app" and "core" package sources, but these will serve as the templates for the `create` package being implemented in #228.  Once that package is ready, I'll remove the boilerplate packages from the repo since they can be trivially reconstructed using `npx create @sdeverywhere`.

Here's what the sample app looks like (it's very minimal but serves as a good starting place for further customization):

<img width="449" alt="image" src="https://user-images.githubusercontent.com/438425/190924498-11c46ae1-2c8d-4dc9-b573-194b77a3255e.png">

**Comment by chrispcampbell** [comment]
Here's the small amount of code that needs to be added to an `sde.config.js` file to make use of the new plugin-config package:

```js
import { configProcessor } from '@sdeverywhere/plugin-config'

export async function config() {
  return {
    // Specify the Vensim model to read
    modelFiles: ['model/sir.mdl'],

    // Read csv files from `config` directory and write to the `generated` directory
    modelSpec: configProcessor({
      config: configDir,
      out: genDir
    }),

    plugins: [
      // ...
    ]
  }
}
```
